### PR TITLE
Return fetch error on response that fails verification or validation

### PR DESCRIPTION
**Pull request by emhane**
Related https://github.com/paradigmxyz/reth/pull/6863.

Returns `FetchEvent::Error` variant if no txns left in `PooledTransactions` response after verification + validation.

### CODE REVIEW

File: crates/net/network/src/transactions/fetcher.rs [hunk 1]
```diff
@@ -867,6 +867,10 @@ impl TransactionFetcher {
                     );
                 }
 
+                if verified_payload.is_empty() {
```
Comment (mattsse) [inline crates/net/network/src/transactions/fetcher.rs:870]: can we add a note here on when this happens?

**Review by mattsse — COMMENTED** [review]
this results in penalization, right?

this also happens, when a peer sends us an empty response?
we can't penalize this because this is a valid response that could just mean: the peer doesn't have any of the requested transactions

**Comment by emhane** [comment]
> this results in penalization, right?
> 
no not yet, it should eventually, issue linked in code comment
> this also happens, when a peer sends us an empty response? we can't penalize this because this is a valid response that could just mean: the peer doesn't have any of the requested transactions

empty response means the peer doesn't have the transactions?...but it should if it announced them...is this by spec or?

**Comment by mattsse** [comment]
> empty response means the peer doesn't have the transactions?...but it should if it announced them...is this by spec or?

>> It is permissible to first announce a transaction via NewPooledTransactionHashes, but then to refuse serving it via PooledTransactions. This situation can arise when the transaction is included in a block (and removed from the pool) in between the announcement and the request.

>> A peer may respond with an empty list iff none of the hashes match transactions in its pool.

> update: in either case is a fetch event error, will make a new one ResponseEmpty "peer didn't serve txns it announced"

that sounds reasonable

**Comment by emhane** [comment]
re-request review @mattsse

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/interfaces/src/p2p/error.rs [hunk 2]
```diff
@@ -59,11 +59,11 @@ impl EthResponseValidator for RequestResult<Vec<Header>> {
     fn reputation_change_err(&self) -> Option<ReputationChangeKind> {
         if let Err(err) = self {
             match err {
-                RequestError::ChannelClosed => None,
-                RequestError::ConnectionDropped => None,
-                RequestError::UnsupportedCapability => None,
+                RequestError::ChannelClosed |
+                RequestError::ConnectionDropped |
+                RequestError::UnsupportedCapability |
```
Comment (mattsse) [inline crates/interfaces/src/p2p/error.rs:64]: please undo changes

**Review by mattsse — APPROVED** [review]
lgtm,

pending restored error.rs

File: crates/interfaces/src/p2p/error.rs (hunk 2 above)
Comment (emhane) [reply crates/interfaces/src/p2p/error.rs:64]: oops, thanks

**Pull request merged**